indico/modules/categories/controllers/display.py

```diff
@@ -52,13 +56,23 @@
 from indico.web.util import jsonify_data
 
 
-CALENDAR_COLOR_PALETTE = [
-    ColorTuple('#1F1100', '#ECC495'),
-    ColorTuple('#0F0202', '#B9CBCA'),
-    ColorTuple('#0D1E1F', '#C2ECEF'),
-    ColorTuple('#000000', '#D0C296'),
-    ColorTuple('#202020', '#EFEBC2')
-]
+def _rgb_to_hex(rgb):
+    return f'#{int(rgb[0] * 255):02x}{int(rgb[1] * 255):02x}{int(rgb[2] * 255):02x}'
+
+
+def _generate_contrast_color(seed):
+    hue = 1.0 / (1.0 + seed % 20)
+    saturation = 1.0 / (1.0 + seed % 4)
+    if saturation < 0.5:
+        saturation += 0.5
+    brightness = 1.0 / (1.0 + seed % 3)
+    if brightness < 0.5:
+        brightness += 0.5
+    r, g, b = colorsys.hsv_to_rgb(hue, saturation, brightness)
+    background_color = _rgb_to_hex((r, g, b))
+    text_color = _rgb_to_hex((0, 0, 0))
```

Comment:
![image](https://github.com/indico/indico/assets/179599/949e50b7-54ad-4943-9a7b-6074e11863e2)

I think for some colors the text need to become white.

Reply:
Hmmmmmm the brightness is guaranteed to be greater or equal to 0.5. In such case I think it's better to use a black color, unless we come up with a different threshold for such value. As an approach I've chosen 0.7. This is the result:

<img width="1070" alt="Screenshot 2024-01-19 at 01 45 37" src="https://github.com/indico/indico/assets/2722756/b1b2fdde-9ad4-450b-83fb-4549193c8d5b">

<img width="1069" alt="Screenshot 2024-01-19 at 01 47 35" src="https://github.com/indico/indico/assets/2722756/fb1f1bc0-30d0-49e1-9108-fd7364e5ea3d">